vpr/src/place/net_cost_handler.cpp

```diff
@@ -88,14 +94,33 @@ static vtr::vector<ClusterNetId, NetUpdateState> bb_updated_before; // [0...clus
  * layer_ts_bb_coord_new are used.
  */
 
-/* [0...cluster_ctx.clb_nlist.nets().size()-1] -> 3D bounding box*/
-static vtr::vector<ClusterNetId, t_bb> ts_bb_coord_new, ts_bb_edge_new;
-/* [0...cluster_ctx.clb_nlist.nets().size()-1][0...num_layers-1] -> 2D bonding box on a layer*/
-static vtr::vector<ClusterNetId, std::vector<t_2D_bb>> layer_ts_bb_edge_new, layer_ts_bb_coord_new;
-/* [0...cluster_ctx.clb_nlist.nets().size()-1][0...num_layers-1] -> number of sink pins on a layer*/
-static vtr::Matrix<int> ts_layer_sink_pin_count;
-/* [0...num_afftected_nets] -> net_id of the affected nets */
-static std::vector<ClusterNetId> ts_nets_to_update;
+struct TSInfo{
+    /* [0...cluster_ctx.clb_nlist.nets().size()-1] -> 3D bounding box*/
+    vtr::vector<ClusterNetId, t_bb> ts_bb_coord_new, ts_bb_edge_new;
+    /* [0...cluster_ctx.clb_nlist.nets().size()-1][0...num_layers-1] -> 2D bonding box on a layer*/
+    vtr::vector<ClusterNetId, std::vector<t_2D_bb>> layer_ts_bb_edge_new, layer_ts_bb_coord_new;
+    /* [0...cluster_ctx.clb_nlist.nets().size()-1][0...num_layers-1] -> number of sink pins on a layer*/
+    vtr::Matrix<int> ts_layer_sink_pin_count;
+    /* [0...num_afftected_nets] -> net_id of the affected nets */
+    std::vector<ClusterNetId> ts_nets_to_update;
+};
+static struct TSInfo ts_info;
+
+/**
+ * @brief This class is used to hide control flows needed to distinguish 2d and 3d placement
+ */
+class BB2D3DControlFlow {
+    bool cube_bb = false; 
+public: 
+    void init(size_t num_nets, bool cube_bb);
+    void get_non_updatable_bb(const ClusterNetId& net);
+    bool is_driver(const t_physical_tile_type_ptr& blk_type, const ClusterPinId& blk_pin);
+    void update_bb(ClusterNetId net_id, t_physical_tile_loc pin_old_loc, t_physical_tile_loc pin_new_loc, bool is_driver); 
+    double get_net_cost(const ClusterNetId& net_id);
+    void set_ts_bb_coord(const ClusterNetId& net_id);
+    void set_ts_edge(const ClusterNetId& net_id);
+};
+static BB2D3DControlFlow bb_2d_3d_control_flow;
```

Comment:
I do not like the name of this class. I would use something like "BBManipulator" or something. Ideally you want to write code in a way that is extensible and self explanatory. BB2D3DControlFlow tells me "There is a problem with the control flow between 2D and 3D BB, and I made this class to encapsulate it" (which honestly is what we are doing), but I would recommend changing the name to something that says "This is a class that will manipulate the BB (whether it is 2D or 3D or any D in the future)".

Reply:
Well put :)

Reply:
BBUpdater?  Or BBandCostUpdater